### PR TITLE
feat(core/caniuse): group by device type

### PR DESCRIPTION
**Pull request by marcoscaceres**
Now looks like: 

Screen Shot 2022-03-14 at 4 41 58 pm

### CODE REVIEW

**Review by sideshowbarker — APPROVED** [review]

File: src/core/caniuse.js [hunk 1]
```diff
@@ -133,18 +141,43 @@ function validateBrowsers({ caniuse }) {
...
   const results = json.result;
-  const out = results.map(({ browser: browserId, version, caniuse }) => {
-    const { name } = BROWSERS.get(browserId);
+  const groups = new Map([
+    ["desktop", []],
+    ["mobile", []],
+  ]);
+  const toBrowserCell = browserCellRenderer(feature);
+  results.reduce(toBrowserCell, groups);
+  const out = [...groups.keys()]
```
Comment (sidvishnoi) [inline src/core/caniuse.js:152]: Can use `.entries()` instead (or shorthand `...groups.filter().map()`)

File: src/core/caniuse.js [hunk 2]
```diff
@@ -133,18 +141,43 @@ function validateBrowsers({ caniuse }) {
...
+    .filter(key => groups.get(key).length)
+    .map(
+      key =>
+        html`<div class="${`caniuse-group`}">
```
Comment (sidvishnoi) [inline src/core/caniuse.js:156]:
```suggestion
        html`<div class="caniuse-group">
```

File: src/core/caniuse.js [hunk 3]
```diff
@@ -133,18 +141,43 @@ function validateBrowsers({ caniuse }) {
...
+          <div class="caniuse-browsers">${groups.get(key)}</div>
+          <div class="caniuse-type"><span>${key}</div>
```
Comment (sidvishnoi) [inline src/core/caniuse.js:158]: Not sure how this works for screen reader users. Should this be like a section heading? (We can move position with CSS)

**Comment by sidvishnoi** [comment]
Is this the intended design on mobile?


I think the more info link should come at bottom at least.

File: src/core/caniuse.js [hunk 4]
```diff
@@ -14,21 +14,29 @@ export const name = "core/caniuse";
...
 
 export const BROWSERS = new Map([
-  ["and_chr", { name: "Android Chrome", path: "chrome" }],
-  ["and_ff", { name: "Android Firefox", path: "firefox" }],
-  ["and_uc", { name: "Android UC", path: "uc" }],
-  ["chrome", { name: "Chrome" }],
-  ["edge", { name: "Edge" }],
-  ["firefox", { name: "Firefox" }],
-  ["ios_saf", { name: "iOS Safari", path: "safari-ios" }],
+  ["and_chr", { name: "Android Chrome", path: "chrome", type: "mobile" }],
```
Comment (sidvishnoi) [inline src/core/caniuse.js:17]: Maybe create two maps? 
(Can merge when testing input for valid browsers)

File: src/styles/caniuse.css.js [hunk 5]
```diff
@@ -4,11 +4,44 @@ const css = String.raw;
 // Prettier ignore only to keep code indented from level 0.
 // prettier-ignore
 export default css`
+
 .caniuse-stats {
   display: flex;
-  flex-wrap: wrap;
+  column-gap: 1cm;
```
Comment (sidvishnoi) [inline src/styles/caniuse.css.js:10]: Maybe use em/rem?

Comment (marcoscaceres) [reply src/styles/caniuse.css.js:10]: Sure, I'll find an equivalent. 1cm looks good to base off.

File: src/core/caniuse.js (hunk 1 above)
Comment (marcoscaceres) [reply src/core/caniuse.js:152]: > Can use .entries() instead 

Also returns an iterator :( 

> or shorthand ...groups.filter().map())

No sure what you mean? move it to another function?

**Comment by marcoscaceres** [comment]
> Is this the intended design on mobile?

Yeah, at least for. We plan to iterate on the design as we get feedback. 

It might make sense to reverse back to column and rows in contained space.

File: src/core/caniuse.js (hunk 3 above)
Comment (marcoscaceres) [reply src/core/caniuse.js:158]: I think it's maybe "labelled-by"?

File: src/core/caniuse.js (hunk 1 above)
Comment (sidvishnoi) [reply src/core/caniuse.js:152]: `out = [...groups].filter(([key, values]) => {})`

File: src/core/caniuse.js (hunk 4 above)
Comment (marcoscaceres) [reply src/core/caniuse.js:17]: oh, you mean DESKTOP_BROWSERS and MOBILE_BROWSERS. I'll take a look.

**Review by sidvishnoi — APPROVED** [review]

**Pull request merged**